plugin/src/test/groovy/com/novoda/test/TestProject.groovy

```diff
@@ -84,15 +84,19 @@ ${project.additionalConfiguration}
         return this
     }
 
-    public T withPlugins(String... plugins) {
-        def formattedPlugins = plugins.collect { plugin ->
-            "id '$plugin'"
-        }.asList()
-
-        this.plugins.addAll(formattedPlugins)
+    public T withPlugin(String plugin, String version = null) {
+        this.plugins.add("id \"$plugin\" ${optionalVersionFrom(version)}")
         return this
     }
 
+    private static String optionalVersionFrom(String version) {
```

Comment:
maybe it's just me, but I would have inlined this as tri-state:
```
this.plugins.add("id '$plugin' ${version?"version '$version'" : ""}")
```

Reply:
Way better 👍  wasn't aware this is possible in groovy